### PR TITLE
Dpd bc

### LINKED ISSUE #953: Decide fate of tunable_slip for Python branch
Is this feature still being used?

### LINKED ISSUE #1381: tunable_slip should be a constraint
pair interaction which only works on one of the particles

### PR DESCRIPTION
**Pull request by fweik**
Fixes #1381. Fixes #953.
See also #1463.

Description of changes:
 - Exposed velocity of constraint particle representation
 - Enabled DPD interaction for constraints, so that they can be used as boundary conditions.

### CODE REVIEW

**Comment by RudolfWeeber** [comment]
Looks good except for missing test and probably 1-2 sentences in the user guide.

**Comment by fweik** [comment]
@RudolfWeeber ready for review.

File: doc/sphinx/system_setup.rst [hunk 1]
```diff
@@ -387,6 +387,12 @@ thermostat acts on the relative velocities between nearest neighbor
 particles. Larger cutoffs including next nearest neighbors or even more
 are unphysical.
 
+Boundary conditions for DPD can be introducees by adding the boundary
```
Comment (KaiSzuttor) [inline doc/sphinx/system_setup.rst]: typo

**Review by KaiSzuttor — APPROVED** [review]
LGTM
